src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -0,0 +1,473 @@
+         [icons/icon :i/audio {:color (colors/theme-colors colors/neutral-100 colors/white)}]]]))])
+
+(defn input-view []
+  [:f>
+   (fn []
+     (fn []
+       [rn/view {:style                         style/input-container
+                 :pointer-events                :box-only
+                 :on-start-should-set-responder (fn [^js e]
+                                                  (when-not @locked?
+                                                    (let [pressed-record-button? (touch-inside-area?
+                                                                                  {:locationX (-> e .-nativeEvent.locationX)
+                                                                                   :locationY (-> e .-nativeEvent.locationY)}
+                                                                                  record-button-area
+                                                                                  false false false false)]
+                                                      (when-not @reviewing-audio?
+                                                        (reset! recording? pressed-record-button?)
+                                                        (when pressed-record-button?
+                                                          ;; TODO: By now we just track recording length, we need to add actual audio recording logic
+                                                          ;; GitHub issue [#14558]: https://github.com/status-im/status-mobile/issues/14558
+                                                          (reset! recording-timer (utils/set-interval #(reset! recording-length-ms (+ @recording-length-ms 500)) 500))))
+                                                      (reset! touch-active? true)))
+                                                  true)
+                 :on-responder-move             (fn [^js e]
+                                                  (when-not @locked?
+                                                    (let [locationX               (-> e .-nativeEvent.locationX)
+                                                          locationY               (-> e .-nativeEvent.locationY)
+                                                          pageX                   (-> e .-nativeEvent.pageX)
+                                                          pageY                   (-> e .-nativeEvent.pageY)
+                                                          moved-to-send-button?   (touch-inside-area?
+                                                                                   {:locationX locationX
+                                                                                    :locationY locationY}
+                                                                                   (send-button-area @ready-to-send?)
+                                                                                   true false false true)
```

Comment:
Why the mix of positional arguments with maps?

Also, passing multiple booleans to functions makes the code hard to read. You can change the `touch-inside-area?` signature, so that the caller code becomes much more readable. Notice that I already changed the example to follow our latest guidelines:

```clojure
(touch-inside-area?
 {:location-x    (oops/oget e "nativeEvent.locationX") 
  :location-y    (oops/oget e "nativeEvent.locationY")
  :ignore-min-y? (if @reviewing-audio? record-button-area record-button-area-big)
  :ignore-max-y? false
  :ignore-min-x? false
  :ignore-min-x? false
  :ignore-max-x? false})
```

Reply:
I don't have a strong reason actually, as I was testing this with UX designers probably I made changes not thinking too much on code readability, but on functionaliy :)

But now is time to improve the former, so thanks for the suggestions, will make those changes.

Reply:
Totally understandable, thanks @briansztamfater

Reply:
Done, I separated it in two maps which I think make more sense:
- The user touch data & params
- The touch area to compare with